From 9a0926013b6dfc4e3f94ff776297c3f4e9b3fc46 Mon Sep 17 00:00:00 2001
From: Aliaksey Kandratsenka <alk@tut.by>
Date: Sat, 1 Nov 2008 13:55:45 +0200
Subject: [PATCH] don't eval recognize_optimized as part of recognition_optimization.rb. It produces meaningless line numbers.

This also easily produces line numbers greater than recognition_optimization.rb have, which causes rcov to trash memory outside of it's coverage counting arrays.
---
 .../routing/recognition_optimisation.rb            |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/actionpack/lib/action_controller/routing/recognition_optimisation.rb b/actionpack/lib/action_controller/routing/recognition_optimisation.rb
index 6c47ced..3b98b16 100644
--- a/actionpack/lib/action_controller/routing/recognition_optimisation.rb
+++ b/actionpack/lib/action_controller/routing/recognition_optimisation.rb
@@ -148,7 +148,7 @@ module ActionController
               end
               nil
             end
-          }, __FILE__, __LINE__
+          }, '(recognize_optimized)', 1
         end
 
         def clear_recognize_optimized!
-- 
1.5.6.5

